airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml

```diff
@@ -0,0 +1,366 @@
+      cursor_field: "updated_at"
+
+  segment_members_stream:
+    type: DeclarativeStream
+    $ref: "#/definitions/base_stream"
+    retriever:
+      $ref: "#/definitions/retriever"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - stream: "#/definitions/segments_stream"
+            parent_key: id
+            partition_field: id
+      record_selector:
+        $ref: "#/definitions/retriever/record_selector"
+        record_filter:
+          type: CustomRecordFilter
+          class_name: source_mailchimp.components.MailChimpRecordFilter
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S%z"
+      cursor_field: "{{ parameters['cursor_field'] }}"
+      start_datetime:
+        type: MinMaxDatetime
+        # TODO: ref to use datetime as optional parameter if possible
+        datetime: "{{ config.get('start_date', day_delta(-500, '%Y-%m-%dT%H:%M:%S.%fZ')) }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      step: "P100Y"
+      cursor_granularity: "PT0.1S"
+    transformations:
+      - type: AddFields
+        fields:
+          - path: ["segment_id"]
+            value: "{{ stream_slice.id }}"
+      - "#/definitions/transformer_remove_empty_fields"
+    $parameters:
+      name: "segment_members"
+      primary_key: "id"
+      path: "/lists/{{ stream_slice.parent_slice.id }}/segments/{{ stream_slice.id }}/members"
+      data_field: "members"
+      cursor_field: "last_changed"
+
+  unsubscribes_stream:
+    type: DeclarativeStream
+    $ref: "#/definitions/base_stream"
+    retriever:
+      $ref: "#/definitions/retriever"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - stream: "#/definitions/campaigns_stream"
+            parent_key: id
+            partition_field: id
+      record_selector:
+        $ref: "#/definitions/retriever/record_selector"
+        record_filter:
+          type: CustomRecordFilter
+          class_name: source_mailchimp.components.MailChimpRecordFilter
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S%z"
+      cursor_field: "{{ parameters['cursor_field'] }}"
+      start_datetime:
+        type: MinMaxDatetime
+        # TODO: ref to use datetime as optional parameter if possible
+        datetime: "{{ config.get('start_date', day_delta(-500, '%Y-%m-%dT%H:%M:%S.%fZ')) }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      step: "P100Y"
+      cursor_granularity: "PT0.1S"
+    $parameters:
+      name: "unsubscribes"
+      primary_key: ["campaign_id", "email_id", "timestamp"]
+      path: "/reports/{{ stream_slice.id }}/unsubscribed"
+      data_field: "unsubscribes"
+      cursor_field: "timestamp"
+
+  email_activity_stream:
```

Comment:
Differences observed:
* The new implementation does not consider `config["campaign_id"]` hence giving 462 records instead of 2

Reply:
campaign id was added as a `hidden` field to config and used in CAT tests only.
I think that CAT are already fast enough not to add this limitation.
Another cons is the structure of YAML:
we need to add another custom component that will choose `type` of `partition_router` based on config["campaign_id"] presence: `SubstreamPartitionRouter` or `ListPartitionRouter`.

I'd prefer not to add this custom component just to cover the test.

Reply:
I didn't know the field was for CATs only. Makes sense to remove it than, thanks!